python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -302,13 +302,41 @@ def _(
     # Join key dtypes are dependent on the schema of the left and
     # right inputs, so these must be translated with the relevant
     # input active.
+    def adjust_literal_dtype(literal: expr.Literal) -> expr.Literal:
+        arrow_type = plc.interop.to_arrow(literal.dtype)
+        if arrow_type == pa.int32():
```

Comment:
I should probably just condition on `plc.types.TypeId` rather than converting `to_arrow`